extensions-core/avro-extensions/src/test/java/org/apache/druid/data/input/AvroStreamInputRowParserTest.java

```diff
@@ -301,7 +301,7 @@ public void testParseSchemaless() throws SchemaValidationException, IOException
 
   static void assertInputRowCorrect(InputRow inputRow, List<String> expectedDimensions, boolean isFromPigAvro)
   {
-    Assert.assertEquals(expectedDimensions, inputRow.getDimensions());
+    Assert.assertEquals(new HashSet<>(expectedDimensions), new HashSet<>(inputRow.getDimensions()));
```

Comment:
@krishnanand5, this would make sense for schemaless tests where the order is non-deterministic when dimension spec is `DimensionsSpec.EMPTY`. On the other hand, when a non-empty schema is specified in the parse spec, order of these dimensions is important and should match the parse spec, so we should assert the lists as-is. 

So I'd suggest passing in `DimensionSpec` to this function and do a conditional assert based on `DimensionsSpec.EMPTY`.